lidar_base.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <stdio.h>
+
+#include "ros/ros.h"
+#include "sensor_msgs/LaserScan.h"
+#include "geometry_msgs/Twist.h"
+
+float min_range,min_range_angle;
+
+ros::Publisher vel_pub;
+geometry_msgs::Twist cmdvel;
+void chatterCallback(const sensor_msgs::LaserScan::ConstPtr& msg)
+{
+ // printf("Position: [%f] [%f]\n", msg->range_min,msg->range_max);
+	min_range=msg->ranges[0];
+	min_range_angle=0;
+	for(int j=0;j<=360;j++) //increment by one degree
```

Comment:
According to the documentation for the [`sensor_msgs/LaserScan`](http://docs.ros.org/api/sensor_msgs/html/msg/LaserScan.html), the minimum and maximum angle for the LIDAR are sent, which should replace the 0->360 degree scan here. Additionally, there is no guarantee that the LIDAR will take one measurement per degree. That information is also published in the message.

Also, the min and max range are published out by the LIDAR (though the angle at which they occur is not)